python/ray/tests/test_multiprocessing.py

```diff
@@ -491,6 +491,22 @@ def f(args):
     with pytest.raises(StopIteration):
         result_iter.next()
 
+    # Repeat of the above with iterators as inputs
```

Comment:
Can we [parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html) this test instead of repeating code? [Here's a good example](https://github.com/ray-project/ray/blob/master/python/ray/serve/tests/test_config.py#L67-L68) from the Ray Serve codebase.

One way to do this here is to add a `use_iter` boolean parameter to this test and configure it to be `True` or `False`. The test signature would look like this:

```python
@pytest.mark.parametrize("use_iter", [True, False])
def test_imap(pool_4_processes, use_iter):
    ...
```

Then, we can check the value of `use_iter` in the test's body. If it's `True`, we can make the input to `imap` and iterator. If it's `False`, we can use a list directly.

Parametrizing tests is nicer than (1) repeating code or (2) embedding for-loops in a test because it allows us to (1) use less code and (2) log clearer errors by specifying what the parameter value is when an error is thrown (a for loop just tells you which line threw an error without telling you what the parameter's value is).

Reply:
Good point! I have done this.